### ISSUE TITLE
MA0042 shouldn't trigger for DbConnection, DbCommand, or DbDataReader

### ISSUE THREAD

**Issue by robv8r** [meziantou/Meziantou.Analyzer#1147]
### Version of the Meziantou.Analyzer NuGet package

3.0.77

### Rule Identifier

MA0042

### Target Framework

net10.0

### C# Language version

C# 14.0

### Description

When creating a custom connection factory that returns a `DbConnection` (whose implementation is a `SqlConnection`), an `await using` should not be required for the DbConnection, DbCommand, or DbDataReader types.

I prefer to use a custom connection factory to avoid a dependency on EF Core (using `IDbContextFactory<TContext>`) while maintaining testability and code coverage.

If there's a better way to accomplish this, I'd be very grateful for an alternative.

### Reproduction Steps

Minimal code:

```c#
using System.Data.Common;
using System.Diagnostics.CodeAnalysis;
using Microsoft.Data.SqlClient;

internal sealed class DataService(IConnectionFactory connectionFactory)
{
    private readonly IConnectionFactory connectionFactory = connectionFactory ?? throw new ArgumentNullException(nameof(connectionFactory));

    public async Task<int> GetDataAsync(CancellationToken cancellationToken = default)
    {
        using var conn = this.connectionFactory.CreateConnection(); // 👈 MA0042 is reported, but shouldn't be.
        await conn.OpenAsync(cancellationToken).ConfigureAwait(false);
        using var cmd = conn.CreateCommand(); // 👈 MA0042 is reported, but shouldn't be.
        cmd.CommandText = "SELECT 1";
        using var reader = await cmd.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false); // 👈 MA0042 is reported, but shouldn't be.

        if (await reader.ReadAsync(cancellationToken).ConfigureAwait(false))
        {
            return reader.GetInt32(0);
        }

        throw new InvalidOperationException("No data returned");
    }
}

internal interface IConnectionFactory
{
    DbConnection CreateConnection();
}

internal sealed class SqlServerConnectionFactory(string connectionString) : IConnectionFactory
{
    private readonly string connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));

    public DbConnection CreateConnection() => new SqlConnection(this.connectionString);
}
```

### Other information

The `csproj` file looks like this:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net10.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <AnalysisLevel>latest-all</AnalysisLevel>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Meziantou.Analyzer" Version="3.0.77">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Data.SqlClient" Version="7.0.1" />
  </ItemGroup>

</Project>
```

**Comment by meziantou** [comment]
The analyzer cannot detect this pattern. There is no way to statically know in `GetDataAsync` what's the type of the connection.

The best approach might be to allow a project to set a list of exclusions, so you can configure certain methods to be ignored and not reported.

**Comment by robv8r** [comment]
For others encountering this issue, it's currently being tracked in #1152
